chef/cookbooks/provisioner/templates/suse/crowbar_register.erb

```diff
@@ -46,9 +46,16 @@ while test $# -gt 0; do
         -h|--help|--usage|-\?) usage ;;
         -v|--verbose) CROWBAR_VERBOSE=1 ;;
         -f|--force) CROWBAR_FORCE=1 ;;
-        -k|--keep-existing-hostname) KEEP_EXISTING_HOSTNAME=1 ;;
         --gpg-auto-import-keys) CROWBAR_AUTO_IMPORT_KEYS=1 ;;
         --no-gpg-checks) CROWBAR_NO_GPG_CHECKS=1 ;;
+        -k|--keep-existing-hostname)
+            if test $# -eq 1; then
```

Comment:
Ah yet I understood what @vuntz said.

Your right, I changed the code a little bit to expect a number as parameter. But I think we should not over engineering the code. When `KEEP_EXISTING_HOSTNAME` is not `1` then the other code will not executed.